prune.go

```diff
@@ -100,6 +100,41 @@ func shouldPrune(oldCount, newCount int, pruneOptions PruneOptions) (bool, strin
 	return false, "transactions have not grown significantly"
 }
 
+// checkMongoSupportsOut verifies that Mongo supports "$out" in an aggregation
+// pipeline. This was introduced in Mongo 2.6
+// https://docs.mongodb.com/manual/reference/operator/aggregation/out/
+func checkMongoSupportsOut(db *mgo.Database) bool {
+	var dbInfo struct {
+		VersionArray []int `bson:"versionArray"`
+	}
+	if err := db.Run(bson.M{"buildInfo": 1}, &dbInfo); err != nil {
+		return false
+	}
+	logger.Debugf("buildInfo reported: %v", dbInfo.VersionArray)
+	if len(dbInfo.VersionArray) < 2 {
+		return false
+	}
+	// Check if we are < 2.6
+	if dbInfo.VersionArray[0] < 2 ||
+		(dbInfo.VersionArray[0] == 2 && dbInfo.VersionArray[1] < 6) {
+		return false
+	}
+	return true
+}
```

Comment:
There is a danger here that this could have a bug causing it to always return false. Then `DBOracle` will never get used but we may not notice (unless we check running times).

What about a unit test which determines if `$out` is supported in another way (maybe by trying to use it and checking for an error) and compares that to what `checkMongoSupportsOut` reports?